app/controllers/chomp_sessions_controller.rb

```diff
@@ -62,4 +62,59 @@ def chomp_params
   def set_chomp_session
     @chomp_session = ChompSession.find_puid(params[:id])
   end
+
+  # input: chomp_session instance
+  # output: one restaurant
+  def generate_restaurant(chomp_session)
+    # algorithm to get recommendation
+    # based on chompsession ID, get all responses
+    # find the lowest budget, convert to integer 1,2,3 or 4
+    # get frequency table of cuisines. Get the highest frequency cuisinse.
+    # If no highest frequency
+    # Get middleground of all location responses and get the highest rated restaurant with the specified cuisine.
+    responses = chomp_session.responses
+    lowest_budget = responses.minimum('budget')
+    restaurant_pricing = determine_pricing(lowest_budget)
+
+    cuisine_arr = []
+    responses.each do |response|
+      response_cuisine = response.cuisine
+      cuisine_arr << response_cuisine.slice!(1, response_cuisine.length) unless response_cuisine == [""]
+    end
+    cuisine_arr.flatten!(1)
+    tallied_result = cuisine_arr.tally
+    most_frequent_cuisine_kv = tallied_result.max_by { |_, value| value }
+
+    # check tie
+    tied_results = tallied_result.select { |_, v| v == most_frequent_cuisine_kv[1] }
```

Comment:
What does the [1] mean? The key-value pair with key "1"?

Reply:
https://ruby-doc.org/core-2.7.3/Enumerable.html#method-i-max_by
Tallied result is a hash containing key value pairs where keys are the cuisine and value is the frequency.
Max_by method returns an **array** containing key in the zeroth index and value on the first index regardless of the fact that there's one or more maximum. Eg. ["Chinese", 5] if there are 5 votes for chinese after 5 responses.

So I have to manually check if there is a tie, by accessing the value at index 1. I think the _kv threw you off. But it's for me know that the data structure is an array but the first value is a key(cuisine) second value is value(frequency).